### PR TITLE
[DT-902] Replace DAC Column with Data Use Column in Data Library Dataset Table

### PR DESCRIPTION
**Pull request by cinyecai**
### Addresses
https://broadworkbench.atlassian.net/browse/DT-902
### Summary
Replaces the DAC column in the Data Library Dataset table with a Data Use column
Column displays Data Use codes (primary and secondary) and has a tooltip explaining each code when you hover over that cell in the table

https://github.com/user-attachments/assets/921ec581-7011-47aa-8594-941e15770b83

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/data_search/DatasetSearchTableConstants.tsx [hunk 1]
```diff
@@ -402,18 +402,50 @@ export const makeDatasetTableHeader = (datasets: DatasetTerm[], selected: number
...
-        data: <OverflowTooltip place={'top'} tooltipText={dataset.dac?.dacName} id={`${dataset.datasetId}-dataset-dac`}>
-          {dataset.dac?.dacName}
-        </OverflowTooltip>,
-        value: dataset.dac?.dacName,
-        id: `${dataset.datasetId}-dac`,
-        style: makeRowStyle(cellWidths.dac),
-        label: `DAC for dataset ${dataset.datasetId}: ${dataset.dac?.dacName}`
-      })
+      cellStyle: makeHeaderStyle(cellWidths.dataUse),
+      cellDataFn: (dataset: DatasetTerm) => {
```
Comment (cinyecai) [inline src/components/data_search/DatasetSearchTableConstants.tsx:410]: This whole function reuses the Data Use logic used on the DAC Datasets table (in the file DACDatasetTableCellData.jsx) to maintain consistency in how each table's Data Use column behaves/looks. It's not as clean as the other cellDataFn's in this file so I'm open to feedback/suggestions on what to do here if this isn't the best approach

**Review by fboulnois — APPROVED** [review]
this looks fantastic, great job 👍

File: src/components/data_search/DatasetSearchTableConstants.tsx (hunk 1 above)
Comment (cinyecai) [reply src/components/data_search/DatasetSearchTableConstants.tsx:410]: Update - Adjusted to using helper function instead

Comment (s-rubenstein) [reply src/components/data_search/DatasetSearchTableConstants.tsx:410]: I think that's okay. I would rather it be different here than duplicate the logic in multiple places.

**Review by s-rubenstein — APPROVED** [review]
LGTM

**Review by snf2ye — APPROVED** [review]

**Pull request merged**